src/components/app/exporters/exportersCommon.ts

```diff
@@ -0,0 +1,25 @@
+import { cloneDeep } from 'lodash';
+import { computed, reactive, ref } from '@vue/composition-api';
+import { VForm } from '@/types/vuetify';
+import store from '@/store';
+import { OutputVendorName } from '@/originalBudgetTrackingApp/commonTypes';
+
+export function setupExporterConfigForm<T extends OutputVendorName>(exporterName: T) {
+  const vForm = ref<VForm>();
+
+  const exporter = reactive(cloneDeep(store.getters.Config.getExporter(exporterName)));
+
+  const validated = ref(true);
+  const changed = ref(false);
+  const readyToSave = computed(() => validated && changed);
+  const submit = async () => {
+    if (vForm.value?.validate()) {
+      await store.dispatch.Config.updateExporter({ name: exporterName, exporter });
```

Comment:
I used *Vuex Actions* becuase it lets us to do some logic related to the commit-dispatch.
But I think now, with TS, we can declare the type we expected for dispatch and we can skip the Action.

Reply:
I am missing some context here...

Reply:
*Vuex* have `getters`, `actions` and `commits`.
You can change the state only by `commit`, but if you want to do some logic, especially if it is an async logic, you need to do that with `action`.
I used `actions` to get a more comfort interface for changing the state, but the right way, if we don't have special logic, to do the change directly with `commit` (AKA `dispatch`)